test/runtests.jl

```diff
@@ -389,6 +389,16 @@ end
     @test t[1,1] == c
 end
 
+@testset "resize!" begin
+    t = StructArray{Pair}(([3, 5], ["a", "b"]))
+    resize!(t, 5)
+    sizehint!(t, 10)
+    @test length(t) == 5
+    p = 1 => "c"
+    t[5] = p
+    @test t[5] == p
+end
+
```

Comment:
Maybe this should be called `"sizehint!"` (or if the two testset are merged, better do all the testing for `resize!` first, then for `sizehint!`). I imagine a good way to test that `sizehint!` worked is something like

```julia
@test @allocated(push!(p, val)) == 0
```

For that to work, you may need `val` to be isbits (eg, `Pair{Int, Int}`).

Reply:
That's a clever idea, I added a new test which does precisely that.